sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1134,6 +1134,31 @@ _pi_queue::_pi_queue(std::vector<ze_command_queue_handle_t> &ComputeQueues,
   CopyCommandBatch.QueueBatchSize = ZeCommandListBatchCopyConfig.startSize();
 }
 
+// Reset signalled command lists in the queue and put them to the cache of
+// command lists.
+pi_result _pi_queue::resetCommandLists() {
+  // We check for command lists that have been already signalled, but have not
+  // been added to the available list yet. Each command list has a fence
+  // associated which tracks if a command list has completed dispatch of its
+  // commands and is ready for reuse. If a command list is found to have been
+  // signalled, then the command list & fence are reset and command list is
+  // returned to the command list cache. All events associated with command list
+  // are cleaned up if command list was reset.
+  std::scoped_lock Lock(this->Mutex);
+  for (auto it = CommandListMap.begin(); it != CommandListMap.end(); ++it) {
```

Comment:
Good catch, but I think I would leave this as is (at least for now) because currently we operate using pi_command_list_ptr_t which is an iterator and resetCommandList inside this loop also accepts an iterator as an input argument.  I mean that stylistic change in this case requires multiple changes in different places.